### PR TITLE
Add basic auth service request functionality

### PR DESCRIPTION
**Pull request by kbirk**
Removed the trusted port approach in favour of https://github.com/unchartedsoftware/pantera-platform-core/pull/408

### CODE REVIEW

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/security/SecurityConfig.java
Comment (bigglesandginger) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/security/SecurityConfig.java:52]: If you want you can totally feel free to delete these Order(2) functions - they are something that should have been deleted a while ago but I was feeling unsure.

**Comment by bigglesandginger** [comment]
A brief review (as I will be unavailable for the next two days) looks good.  I approve but have not tested to see if the secure route vs unsecure is working (one should be able to use the app normally, not able to cURL an app api endpoint without creds, and equally not be able to use 3001 with the client but be able to cURL without creds... .

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/TerariumApplication.java
Comment (dvince2) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/TerariumApplication.java:24]: Not a deal breaker but in general we're now trying to avoid `@Autowired` in favour of a `@RequiredArgsConstructor`

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/configuration/Config.java
Comment (dvince2) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/configuration/Config.java:49]: Before merging to main we should have this defined in `application.properties`

**Review by dvince2 — APPROVED** [review]

**Review by dvince2 — APPROVED** [review]

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/security/SecurityConfig.java
Comment (kbirk) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/security/SecurityConfig.java]: Done!

**Pull request merged**
